addons/event/models/event_registration.py

```diff
@@ -93,18 +95,16 @@ def _compute_date_closed(self):
                 else:
                     registration.date_closed = False
 
-    @api.constrains('event_id', 'state')
-    def _check_seats_limit(self):
-        for registration in self:
-            event_id = registration.event_id
-            if event_id.seats_limited and event_id.seats_max and event_id.seats_available < (1 if registration.state == 'draft' and event_id.auto_confirm else 0):
-                raise ValidationError(_('There are no more available seats for %s. Raise the limit or remove some other confirmed registrations first.', event_id.name))
+    @api.constrains('event_id', 'event_ticket_id', 'state')
+    def _check_seats_availability(self):
```

Comment:
As we already have a constraint at event level and ticket level on registration_ids, is this one really necessary ? Isn't it double computation ?